mcsema/BC/Function.cpp

```diff
@@ -260,6 +289,132 @@ static llvm::BasicBlock *GetOrCreateBlock(TranslationContext &ctx,
   return block;
 }
 
+static void LiftExceptionFrameLP(TranslationContext &ctx,
+                                 const NativeFunction *cfg_func) {
+
+  if (cfg_func->eh_frame.size() > 0) {
+
+    // The personality function `__gxx_personality_v0` is lifted as global variable. Erase the
+    // variable before declaring it as the function.
+    if (auto personality_func_var = gModule->getGlobalVariable("__gxx_personality_v0")) {
+      personality_func_var->eraseFromParent();
+    }
+
+    auto personality = gModule->getFunction("__gxx_personality_v0");
+
+    if (personality == nullptr) {
+      personality = llvm::Function::Create(
+          llvm::FunctionType::get(llvm::Type::getInt32Ty(*gContext), true),
+          llvm::Function::ExternalLinkage, "__gxx_personality_v0", gModule);
+    }
+
+    auto lifted_func = gModule->getFunction(cfg_func->lifted_name);
+    lifted_func->addFnAttr(llvm::Attribute::UWTable);
+    lifted_func->addFnAttr(llvm::Attribute::OptimizeNone);
+    lifted_func->removeFnAttr(llvm::Attribute::NoUnwind);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+    lifted_func->setPersonalityFn(personality);
+#endif
+  }
+
+  for (auto &entry : cfg_func->eh_frame) {
+    if (entry->lp_ea == 0) continue;
+
+    if (entry->action == 0) {
+      std::stringstream ss;
+      ss << "cleanup_landingpad_" << std::hex << entry->lp_ea;
+      llvm::BasicBlock *landing_bb = llvm::BasicBlock::Create(
+          *gContext, ss.str(), ctx.lifted_func);
+      llvm::IRBuilder<> ir(landing_bb);
+      auto exn_type = llvm::StructType::get(llvm::Type::getInt8PtrTy(*gContext),
```

Comment:
Maybe add a comment here saying that the first element is a FOO and the second one is a BAR?